### PR TITLE
SEO: add structured data

### PR DESCRIPTION
**Pull request by lewisakura**
This helps improve SEO by adding some structured data tags to the page. This adds a `WebSite` schema (which makes it easier for Google to know the site name and canonical in the event that it gets confused) and, specifically for the FAQ page, the `FAQPage` structured data which *can* be rendered as those FAQ dropdowns in Google search. Even if they don't get rendered, it contributes to SEO greatly (according to, well, Google).

### CODE REVIEW

File: src/pages/faq.astro [hunk 1]
```diff
@@ -5,6 +5,22 @@ import Layout from "layouts/Layout.astro";
 import { getSortedFaq } from "scripts/collections";
 
 const faq = await getSortedFaq();
+
+// this sucks, but it's the only way to generate the right data for structured data
+// regardless, this only runs at prerender time so it's okay
+const rawFaqData = await Astro.glob("../content/faq/*.md");
```
Comment (Vendicated) [inline src/pages/faq.astro:11]: why is this needed? so you can obtain raw html?

Comment (lewisakura) [reply src/pages/faq.astro:11]: correct - Astro collections don't give you access to the raw HTML content (only the component) so we have to use Astro's glob to import them and do it that way

File: src/layouts/Layout.astro [hunk 2]
```diff
@@ -12,6 +12,14 @@ export interface Props {
...
 
 const { title, description, opengraph } = Astro.props;
+
+const structuredData = JSON.stringify({
+    "@context": "https://schema.org",
+    "@type": "WebSite",
+    name: "Vencord",
+    alternateName: "VC",
+    url: Astro.site,
+});
```
Comment (Vendicated) [inline src/layouts/Layout.astro:22]: is this really necessary here? It seems obsolete as this info is already obtainable via og tags

Comment (lewisakura) [reply src/layouts/Layout.astro:22]: surprisingly, yes! if you have it in open graph and structured data it increases google's reliability at actually detecting your site information (because at the minute it doesnt know the site is called "Vencord", its just "vencord.dev")

**Comment by Vendicated** [comment]
looking a bit into this cause i didn't know about it until now and all the google examples put this in the head instead of body. Is that of relevance? Maybe we should do the same

**Comment by lewisakura** [comment]
> looking a bit into this cause i didn't know about it until now and all the google examples put this in the head instead of body. Is that of relevance? Maybe we should do the same

https://developers.google.com/search/docs/appearance/structured-data/intro-structured-data#supported-formats

can go in both, as long as it exists.
